### DISCUSSION TITLE
TypeError: Key must be one of type KeyObject or CryptoKey.

### DISCUSSION THREAD

**Discussion by Roslam24** [panva/jose#427] [Q&A]
I wanna ask, i try to sign JWT using method SignJWT and use same as documentation shows me. but i got and error,  it's say ``` TypeError: Key must be one of type KeyObject or CryptoKey.``` cant you give me the clue for this issue? thanks

**Comment by panva** [comment]
- https://github.com/panva/jose/issues/210
- https://github.com/panva/jose/blob/v4.8.3/docs/types/types.KeyLike.md
- https://github.com/panva/jose/blob/main/docs/classes/jwt_sign.SignJWT.md#readme

**Comment by duinness** [comment]
This is 6 months old, but I had this same issue. The links provided are very informative but it wasn't clear to me what I was doing wrong.
The reason I got this error is because I wasn't exactly following the example in that I was skipping the "TextEncoder().encode" step. 

This stack overflow post was very helpful. 
https://stackoverflow.com/questions/72385826/create-a-nodejs-keyobject-from-a-string

> The only thing you're missing is setting the JOSE protected header with an appropriate symmetric JWS algorithm (alg).
> 
> The [example](https://github.com/panva/jose/blob/main/docs/classes/jwt_sign.SignJWT.md#readme) show how that's done and the appropriate symmetric algorithm is HS256.

> import * as jose from "jose";
> 
> const secret = new TextEncoder().encode(
>   "Swe4g7c?UBm5Nrd96vhsVDtkyJFbqKMTm!TMw5BDRLtaCFAXNvbq?s4rGKQSZnUP"
> );
> 
> const jwt = await new jose.SignJWT({ foo: "bar" })
>   .setProtectedHeader({ alg: "HS256" })
>   .sign(secret);
> console.log(jwt); // eyJhbGciOiJIUzI1NiJ9.eyJmb28iOiJiYXIifQ.iPFY1ibZc5dTBzRD46ma-Du0avf20nYKtQQsgnyf7ZM

**Reply by mabels** [reply]
Hopefully this will fix it:

https://github.com/panva/jose/pull/769
